zerver/actions/user_settings.py

```diff
@@ -447,3 +451,25 @@ def do_change_user_setting(
         # not deleted every previously synced draft - to do that use the DELETE
         # endpoint.
         Draft.objects.filter(user_profile=user_profile).delete()
+
+    # if user changes "Display my availability to other users when online" setting,
+    # then quickly show/hide them to other uses via do_update_user_presence()
+    # which calls send_event. see https://github.com/zulip/zulip/issues/21108
+    if setting_name == "presence_enabled":
+
+        # remove all other presence indicators for this user
+        if setting_value is False:
+            UserPresence.objects.filter(user_profile_id=user_profile.id).delete()
+
+        # use OFFLINE_THRESHOLD_SECS to force the user to appear INACTIVE (not just idle)
+        timesmear = 0 if setting_value else (settings.OFFLINE_THRESHOLD_SECS + 1)
```

Comment:
I'd prefer an if/else here, because that will ensure the test coverage system verifies that we run both code paths in our tests.